File: River II/RIVER II.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;


class Solution
{
    static void Main(string[] args)
    {
        int r1 = int.Parse(Console.ReadLine());

        for (int i = r1 - 1; i > 0; i--)
        {
            int check = getnextnumber(i);
            if (check == r1)
            {
                Console.WriteLine("YES");
                break;
            }
            else if (i == 1)
            Console.WriteLine("NO");
        };

    }

static int getnextnumber(int input)
    {
        int result = input;

        while (input > 0)
        {
            result += input % 10;
            input -= input % 10;
            input /= 10;
        }
        return result;
    }
}

File: Easy/River II/RIVER II.py
import sys
import math

def getnextnumber(i):
    result = i
    while i > 0:
        result += i % 10
        i -= i % 10
        i /= 10
    return result

r_1 = int(input())
for i in range(r_1,0,-1):
    check = getnextnumber(i)
    if (check == r_1):
        print("YES")
        break
    elif (i == 1):
        print("NO")